search/controllers/api/__init__.py

```diff
@@ -142,39 +155,44 @@ def classic_query(params: MultiDict) \
         raise BadRequest("Either a search_query or id_list must be specified"
                          " for the classic API.")
 
+
     if raw_query:
         # migrate search_query -> query variable
-        params['query'] = raw_query
-        del params['search_query']
-
-        # pass to normal search, which will handle parsing
-        data, _, _ = search(params)
-
-    if id_list and not raw_query:
+        phrase = parse_classic_query(raw_query)
+        # TODO: add support for order, size, page_start
+        query = ClassicAPIQuery(phrase=phrase)
+        
+        # pass to search indexer, which will handle parsing
+        document_set: DocumentSet = index.SearchSession.search(query)
+        data: SearchResponseData = {'results': document_set, 'query': query}
+        logger.debug('Got document set with %i results',
+                        len(document_set['results']))
+
+        if id_list: # and raw_query
+            results = [paper for paper in document_set.results
+                        if paper.paper_id in id_list or paper.paper_id_v in id_list]
+            data = {
+                'results' : DocumentSet(results=results, metadata=dict()), # TODO: Aggregate search metadata
+                'query' : query
+            }
+    
+    elif id_list: # and not raw_query
         # Process only id_lists.
         # Note lack of error handling to implicitly propogate any errors.
         # Classic API also errors if even one ID is malformed.
         papers = [paper(paper_id) for paper_id in id_list]
 
-        data, _, _ = zip(*papers)
-        results = [paper['results'] for paper in data]   # type: ignore
-        data = {
-            'results' : dict(results=results, metadata=dict()), # TODO: Aggregate search metadata
-            'query' : APIQuery() # TODO: Specify query
-        }
-
-    elif id_list and raw_query:
-        # Filter results based on id_list
-        results = [paper for paper in data['results']['results']
-                       if paper.paper_id in id_list or paper.paper_id_v in id_list]
+        papers, _, _ = zip(*papers) # type: ignore
```

Comment:
The inferred type for ``papers`` looks like ``List[Tuple[Dict[str, Any], int, Dict[str, Any]]]`` (L157/183). But then it looks like you are assigning ``List[Dict[str, Any]]`` to it. Rather than suppress the type error (which is truly an error), try using a different name.